modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2PortalActivation.testcase

```diff
@@ -0,0 +1,403 @@
+			CP2NavigationMenu.goToProductActivation();
+
+			CP2NavigationMenu.clickOnProductButton(product = "Portal");
+		}
+
+		task ("Click on Filter and Select a Filter type") {
+			CustomerPortal.clickAtButton(text = "Filter");
+
+			WaitForVisible(
+				key_text = "Start Date",
+				locator1 = "CustomerPortalSite#ANY_BUTTON");
+
+			CustomerPortal.clickAtButton(text = "Start Date");
+		}
+
+		task ("Click Start Date and fill in the field and assert the results of the label") {
+			CP2NavigationMenu.typeDate(
+				data = "Jun 7 2022",
+				field = "On or After",
+				index = "4",
+				index1 = "1");
+
+			Click(
+				key_text = "Redacted",
+				locator1 = "CP2ActivationKeys#RESPONSIVE_TABLE");
+
+			CP2ActivationKeys.assertKeyDetailsStatusElements(startDate = "Jun 7, 2022");
+		}
+	}
+
+	@description = "Automation ID: LPS-151350 | Test Summary: Verify if after the user click on the Apply button, the list of activation keys matches the selected criteria."
+	@ignore = "true"
+	test VerifyPortalActivationApplyButton {
+
+		// Needs refactor since it depends on real projects Portals account
+
+		task ("Signs in OKTA to update and go to Customer Portal") {
+			CustomerPortal.loginOktaToUpdateCP();
+
+			Navigator.openURL();
+
+			ApplicationsMenu.gotoSite(key_site = "${testSiteName}");
+		}
+
+		task ("Select a project") {
+			ApplicationsMenu.gotoSite(key_site = "${testSiteName}");
+
+			CustomerPortal.search(searchItem = "NASA ICE");
+
+			CustomerPortal.enterOnProject(title = "NASA ICE");
+		}
+
+		task ("Go to Product Activation and Product page") {
+			CP2NavigationMenu.goToProductActivation();
+
+			CP2NavigationMenu.clickOnProductButton(product = "Portal");
+		}
+
+		task ("Click on Filter and Select a Filter type") {
+			CustomerPortal.clickAtButton(text = "Filter");
+
+			CustomerPortal.clickAtButton(text = "Key Type");
+
+			Click.clickAt(
+				filter = "On-Premise",
+				index = "1",
+				locator1 = "CP2NavigationMenu#FILTER_CHECKBOX");
+		}
+
+		task ("Assert if the list reflect the filter") {
+			CustomerPortal.clickAtButton(text = "Apply");
+
+			AssertElementPresent(
+				locator1 = "CP2NavigationMenu#FILTER_RESULT",
+				result = "On-Premise");
+
+			Click(
+				key_text = "Redacted",
+				locator1 = "CP2ActivationKeys#RESPONSIVE_TABLE");
+
+			CP2ActivationKeys.assertKeyDetailsServerElements(keyType = "On-Premise");
+		}
+	}
+
+	@description = "Automation ID: LPS-151348 | Test Summary: Verify if the user can search a IP Addresses."
+	@ignore = "true"
+	@priority = "3"
+	test VerifyUserCanSearchIPAddresses {
+
+		// Needs refactor since it depends on real projects Portals account
+
+		task ("Signs in OKTA to update and go to Customer Portal") {
+			CustomerPortal.loginOktaToUpdateCP();
+
+			Navigator.openURL();
+
+			ApplicationsMenu.gotoSite(key_site = "${testSiteName}");
+		}
+
+		task ("Select a project and go to activation page") {
+			CustomerPortal.enterOnProject(title = "NASA ICE");
```

Comment:
@styller1 Why this test doesnt need the  macro call 

CustomerPortal.search(searchItem = "NASA ICE");

before CustomerPortal.enterOnProject(title = "NASA ICE");

Is it different?